osu.Game/Beatmaps/IO/OszArchiveReader.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+using Ionic.Zip;
+using osu.Game.Beatmaps.Formats;
+
+namespace osu.Game.Beatmaps.IO
+{
+    public class OszArchiveReader : ArchiveReader
+    {
+        static OszArchiveReader()
+        {
+            AddReader<OszArchiveReader>((storage, path) =>
+            {
+                using (var stream = storage.GetStream(path))
+                {
+                    if (!ZipFile.IsZipFile(stream, false))
+                        return false;
+                    using (ZipFile zip = ZipFile.Read(stream))
+                        return zip.Entries.Any(e => e.FileName.EndsWith(".osu"));
+                }
+            });
+        }
+    
+        private ZipFile Archive { get; set; }
+        private string[] Beatmaps { get; set; }
+        private Beatmap FirstMap { get; set; }
+    
+        public OszArchiveReader(Stream archive)
+        {
+            Archive = ZipFile.Read(archive);
+            Beatmaps = Archive.Entries.Where(e => e.FileName.EndsWith(".osu"))
```

Comment:
I didn't want to necessarily have to *parse* beatmaps at this stage, just get a list of their files.